include/blmc_robots/common_header.hpp

```diff
@@ -141,6 +143,28 @@ void print_vector(std::string v_name, Eigen::Ref<Eigen::VectorXd> v)
     rt_printf("]\n");
 }
 
+/**
+ * @brief This small structure is used for reading the calibration parameters
+ * for the calibrations demos.
+ * 
+ * @tparam ROBOT_TYPE 
+ */
+template<class ROBOT_TYPE>
+struct ThreadCalibrationData{
+  ROBOT_TYPE* robot;
+  Eigen::Vector2d joint_index_to_zero;
+
+  ThreadCalibrationData(ROBOT_TYPE* robot_in){
```

Comment:
is it a good idea to pass a plain pointer? the user may pass a pointer to robot_in and then destroy the robot_in-object, and the member of your class will point into the void. i think a shared pointer would be safer.